modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/sql/GridSqlQuerySplitter.java

```diff
@@ -702,12 +718,15 @@ private static boolean splitSelectExpression(List<GridSqlElement> mapSelect, Lis
      * @param el Expression.
      * @return {@code true} If expression contains aggregates.
      */
-    private static boolean hasAggregates(GridSqlElement el) {
-        if (el instanceof GridSqlAggregateFunction)
+    private static boolean hasAggregates(GridSqlElement el, boolean distinct) {
```

Comment:
Where is javadoc param description for `distinct`? Also I don't like the naming of this new parameter: even when looking at the code I can't clearly understand what this parameter means. It would be even better to move the whole expression after `el instanceof GridSqlAggregateFunction &&` to a separate method with clear name and javadoc.